### DISCUSSION TITLE
Can't reproduce TweedieDevianceScore for power=1 using formula stated in docs

### DISCUSSION THREAD

**Discussion by EvanZ** [Lightning-AI/torchmetrics#2319] [Q&A] [answered]
Here's an example with output:

```
deviance = TweedieDevianceScore(power=1)
yhat = torch.tensor([0.01, 0.01, 0.1, 0.2, 0.001, 0.00001])
y = torch.tensor([0, 0, 0, 1, 0, 0])
deviance(yhat, y)
>tensor(0.3101)
```

```
def tweedie_deviance_score(yhat, y):
  return 2 * (y * torch.log(y/yhat) + yhat - y)

tweedie_deviance_score(yhat, y)
>tensor([   nan,    nan,    nan, 1.6189,    nan,    nan])
```

As far as I can tell the formula in my function is the same as what is stated in the docs for `power=1` (loss Poisson distribution). What am I missing here?

**Comment by Borda** [comment, marked as answer]
adding a few more prints to your function:
```py
def tweedie_deviance_score(yhat, y):
    print(y/yhat)
    print(torch.log(y/yhat))
    print(y * torch.log(y/yhat))
    return 2 * (y * torch.log(y/yhat) + yhat - y)
```
and seems the problem is with `inf` as you can see:
```
tensor([0., 0., 0., 5., 0., 0.])
tensor([  -inf,   -inf,   -inf, 1.6094,   -inf,   -inf])
tensor([   nan,    nan,    nan, 1.6094,    nan,    nan])
```
